asv_bench/benchmarks/utils/common.py

```diff
@@ -563,3 +563,32 @@ def prepare_io_data(test_filename: str, data_type: str, shapes: list):
         df.to_csv(test_filenames[shape_id], index=False)
 
     return test_filenames
+
+
+def prepare_io_data_parquet(test_filename: str, data_type: str, shapes: list):
+    """
+    Prepare data for IO tests with caching.
+
+    Parameters
+    ----------
+    test_filename : str
+        Unique file identifier that is used to distinguish data
+        for different tests.
+    data_type : {"int", "str_int", "true_false_int"}
+        Type of data generation.
+    shapes : list
+        Data shapes to prepare.
+
+    Returns
+    -------
+    test_filenames : dict
+        Dictionary that maps dataset shape to the file on disk.
+    """
+    test_filenames = {}
+    for shape in shapes:
+        shape_id = get_shape_id(shape)
+        test_filenames[shape_id] = f"{test_filename}_{shape_id}.parquet"
```

Comment:
If multiple `prepare_io_data_parquet` are called with different `data_type`, the file names would conflict, so perhaps we could add `data_type` to the file name.

Reply:
Good point - this also applies to `csv` format.